lib/open_api_spex/cast_parameters.ex

```diff
@@ -31,11 +31,16 @@ defmodule OpenApiSpex.CastParameters do
     Plug.Conn.fetch_cookies(conn).req_cookies
   end
 
+  # Matching is done using the downcase version of the expected_names but the original
+  # version of the expected_names is kept
   defp get_params_by_location(conn, :header, expected_names) do
+    expected_names = Map.new(expected_names, & {String.downcase(&1), &1})
```

Comment:
I found this code a bit terse to understand initially. How about we introduce a new variable rather than shadow `expected_names`, since it is changing from a list of strings to a map, and use a `for` comprehension to map the headers?


```elixir
  # Matching is done using the downcase version of the expected_names but the original
  # version of the expected_names is kept
  def get_params_by_location(conn, :header, expected_names) do
    name_lookup = Map.new(expected_names, & {String.downcase(&1), &1})

    for {key, value} <- conn.req_headers,
        property_name = name_lookup[String.downcase(key)],
        into: %{}, do: {property_name, value}
  end
```

Reply:
Looks good 👌